UnitTests/Subscription/SubscriptionPagesUseSubscriptionFeatureTests.swift

```diff
@@ -966,6 +966,90 @@ final class SubscriptionPagesUseSubscriptionFeatureTests: XCTestCase {
         XCTAssertTrue(tokenResponse.isEmpty)
         XCTAssertPrivacyPixelsFired([])
     }
+
+    func testSubscriptionUpgradeNotificationSentWhenSubscriptionSelectedSuccessFromFreemium() async throws {
+        // Given
+        ensureUserUnauthenticatedState()
+        XCTAssertEqual(subscriptionEnvironment.purchasePlatform, .appStore)
+        XCTAssertFalse(accountManager.isUserAuthenticated)
+
+        storePurchaseManager.hasActiveSubscriptionResult = false
+        storePurchaseManager.mostRecentTransactionResult = nil
+
+        authService.createAccountResult = .success(CreateAccountResponse(authToken: Constants.authToken,
+                                                                         externalID: Constants.externalID,
+                                                                         status: "created"))
+        authService.getAccessTokenResult = .success(AccessTokenResponse(accessToken: Constants.accessToken))
+        authService.validateTokenResult = .success(Constants.validateTokenResponse)
+        storePurchaseManager.purchaseSubscriptionResult = .success(Constants.mostRecentTransactionJWS)
+        subscriptionService.confirmPurchaseResult = .success(ConfirmPurchaseResponse(email: Constants.email,
+                                                                                     entitlements: Constants.entitlements,
+                                                                                     subscription: SubscriptionMockFactory.subscription))
+
+        let mockfreemiumDBPUserStateManager = MockFreemiumDBPUserStateManager()
+        mockfreemiumDBPUserStateManager.didActivate = true
+
+        let mockNotificationCenter = MockNotificationCenter()
```

Comment:
Not an issue, but you can check for fired notifications without mocking NC with just expectations, see: https://github.com/duckduckgo/macos-browser/blob/589c6dd9d99ec74b60ecdb978b58f56f2eb74523/UnitTests/Subscription/SubscriptionPagesUseSubscriptionFeatureTests.swift#L312

Reply:
Nice! I’ll use that going forward.